core/tests/puppeteer-acceptance-tests/user-utilities/translation-admin-utils.js

```diff
@@ -0,0 +1,177 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Translation admin users utility file.
+ */
+
+const baseUser = require(
+  '../puppeteer-testing-utilities/puppeteer-utils.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/test-constants.js');
+const ContributorDashboardAdminUrl =
+  testConstants.URLs.ContributorDashboardAdmin;
+const { showMessage } = require(
+  '../puppeteer-testing-utilities/show-message-utils.js');
+
+const translationRightValue = 'string:translation';
+const usernameMethodValue = 'username';
+const roleMethodValue = 'role';
+
+// "View Contributor Dashboard Users" form elements.
+const viewContributorMethodSelect =
+  'select#view-contributor-method-select';
+const viewContributerUsernameInput =
+  'input#view-contributor-username-input';
+const viewContributorCategorySelect =
+  'select#view-contributor-category-select';
+const viewContributorLanguageSelect =
+  'select#view-contributor-language-select';
+const viewContributorSubmitButton =
+  'button#view-contributor-submit-button';
+const viewContributorLanguageResult =
+  '.e2e-test-translation-reviewer-language';
+const viewLanguageRoleUserResult =
+  '.e2e-test-reviewer-roles-result';
+
+// "Add Contribution Rights" form elements.
+const addContributorUsernameInput =
+  'input#add-contribution-rights-user-input';
+const addContributonRightsCategorySelect =
+  'select#add-contribution-rights-category-select';
+const addContributonRightsLanguageDropdown =
+  'select#add-contribution-rights-language-select';
+const addontributionRightsSubmitButton =
+   'button#add-contribution-rights-submit-button';
+
+// "Remove Contribution Rights" form elements.
+const revokeContributorUsernameInput =
+  'input#revoke-contribution-rights-user-input';
+const revokeContributonRightsCategorySelect =
+  'select#revoke-contribution-rights-category-select';
+const revokeContributonRightsLanguageSelect =
+  'select#revoke-contribution-rights-language-select';
+const revokeContributionRightsSubmitButton =
+  'button#revoke-contribution-rights-submit-button';
+
+module.exports = class TranslationAdmin extends baseUser {
+  /**
+   * Function for navigating to the contributor dashboard admin page.
+   */
+  async navigateToContributorDashboardAdminPage() {
+    await this.goto(ContributorDashboardAdminUrl);
+  }
+
+  /**
+   * Function for assigning a translation right to a user.
+   * @param {string} username - the username of the user.
+   * @param {string} languageValue - the language to assign as an option value.
+   */
+  async assignTranslationRights(username, languageValue) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, translationRightValue);
+    await this.select(addContributonRightsLanguageDropdown, languageValue);
+    await this.clickOn(addontributionRightsSubmitButton);
+
+    // Wait a moment for changes to take effect.
+    await this.page.waitForTimeout(1000);
```

Comment:
please read the comment I left for waitForTimeout.

Reply:
Replaced with [waitForNetworkIdle](https://pptr.dev/api/puppeteer.page.waitfornetworkidle)